app/models/post.rb

```diff
@@ -1,2 +1,13 @@
 class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User', foreign_key: :author_id
+  has_many :comments, foreign_key: :post_id
+  has_many :likes, foreign_key: :post_id
+
+  def user_posts_count
+    author.update_column(:posts_counter, author.posts.count)
+  end
```

Comment:
- [ ] [Optional] It would be better to use increment! as increment! method doesn't want us to save it manually. why it is better you can see [here](https://stackoverflow.com/questions/11499110/increment-vs-1).
`example only` 
```ruby
def update_post_counter
  author.increment!(:posts_counter)
end
```